akka-actor/src/main/scala/akka/japi/JavaAPI.scala

```diff
@@ -217,6 +217,32 @@ object Option {
  */
 object Util {
 
+  /**
+   * Returns `null`, typed to `Class[T]`.
+   *
+   * <p>Provided for the methods in Akka that have a `Class` parameter used solely for its type
+   * parameter, like `Flow.of`.  As such, it is recommended to be used inline, additionally because
+   * it will allow the type parameter will be inferred, like so:
+   * {{{ Flow.of(Util.nullClassValue()) }}}
+   * but, if necessary, the type parameter can also be specified:
+   * {{{ Flow.of(Util.<Optional<String>>nullClassValue()) }}}
+   *
+   * <p>The motivation for such a method is the fact that Java's `.class` syntax only supports
+   * non-generic types, for example `String.class` to obtain a `Class<String>` value, and cannot
+   * directly return, for instance, a value of type `Class<Optional<String>`.
+   *
+   * <p>The workaround in Java is:
+   * {{{
+   * @SuppressWarnings("unchecked")
+   * Flow.of((Class<Optional<String>>) (Class<?>) Optional.class)
```

Comment:
`Optional.of("").getClass`

Reply:
No, `getClass()` returns `Class<?>`.  So you get:

```
[error] /d/akka/akka-actor-tests/src/test/java/akka/japi/JavaAPITestBase.java:90:1: incompatible types: inferred type does not conform to equality constraint(s)
[error]     inferred: java.util.Optional<java.lang.String>
[error]     equality constraints(s): java.util.Optional<java.lang.String>,capture#1 of ? extends java.util.Optional
[error]     final Flow<Optional<String>> f1 = Flow.of(Optional.of("").getClass());
```